File: be-core/src/main/java/com/realman/becore/dto/account/AccountInfo.java
package com.realman.becore.dto.account;

import java.time.LocalDateTime;

import com.realman.becore.dto.enums.EAccountStatus;
import com.realman.becore.dto.enums.EGender;
import com.realman.becore.dto.enums.EItimacyLevel;
import com.realman.becore.dto.enums.EProfessional;
import com.realman.becore.dto.enums.ERole;

public interface AccountInfo {
    void setFirstName(String firstName);
    void setLastName(String lastName);
    void setThumbnailUrl(String thumbnailUrl);
    void setGender(EGender gender);
    void setDob(LocalDateTime dob);
    void setPhone(String phone);
    void setAddress(String address);
    void setStatus(EAccountStatus status);
    void setRole(ERole role);

    void setBranchName(String branchName);
    void setBranchAddress(String branchAddress);

    void setProfessional(EProfessional professional);
    void setAverageRating(Double averageRating);

    void setItimacyLevel(EItimacyLevel itimacyLevel);

    String getFirstName();
    String getLastName();
    String getThumbnailUrl();
    EGender getGender();
    LocalDateTime getDob();
    String getPhone();
    String getAddress();
    EAccountStatus getStatus();
    ERole getRole();

    String getBranchName();
    String getBranchAddress();

    EItimacyLevel getItimacyLevel();

    EProfessional getProfessional();
    Double getAverageRating();
}

File: realmen_customer_application/lib/models/account_info.dart
import 'dart:core';

class AccountInfoModel {
  int? accountId;
  int? branchId;
  String? thumbnailUrl;
  String? firstName;
  String? lastName;
  String? phone;
  String? address;
  String? dob;
  String? gender;
  String? status;
  String? branchName;
  String? branchAddress;
  String? itimacyLevel;
  String? professional;
  int? average;
  String? role;

  AccountInfoModel({
    this.accountId,
    this.branchId,
    this.thumbnailUrl,
    this.firstName,
    this.lastName,
    this.phone,
    this.address,
    this.dob,
    this.gender,
    this.status,
    this.branchName,
    this.branchAddress,
    this.itimacyLevel,
    this.professional,
    this.average,
    this.role,
  });

  factory AccountInfoModel.fromJson(Map<String, dynamic> json) {
    return AccountInfoModel(
      accountId: json['value']['accountId'],
      branchId: json['value']['branchId'],
      thumbnailUrl: json['value']['thumbnailUrl'],
      firstName: json['value']['firstName'],
      lastName: json['value']['lastName'],
      phone: json['value']['phone'],
      address: json['value']['address'],
      dob: json['value']['dob'],
      gender: json['value']['gender'],
      status: json['value']['status'],
      branchName: json['value']['branchName'],
      branchAddress: json['value']['branchAddress'],
      itimacyLevel: json['value']['itimacyLevel'],
      professional: json['value']['professional'],
      average: json['value']['average'],
      role: json['value']['role'],
    );
  }
}
